### ISSUE TITLE
VMRule not found

### ISSUE THREAD

**Issue by r-mironov** [VictoriaMetrics/operator#524]
Hello!

I'm testing [victoria-metrics-k8s-stack](https://github.com/VictoriaMetrics/helm-charts/tree/master/charts/victoria-metrics-k8s-stack) and it works great. 

But I've faced with one problem, when I create resource `prometheusrules`, operator converts it and creates `VMRule` and it works fine. After that I try to delete `prometheusrules` and expect that `VMRule` will deleted correctly. 

VMRule resource was deleted, however victoria-metrics-operator writes these messages to log:
```
{
  "level": "error",
  "ts": 1662463379.7290535,
  "msg": "Reconciler error",
  "controller": "vmrule",
  "controllerGroup": "operator.victoriametrics.com",
  "controllerKind": "VMRule",
  "vMRule": {
    "name": "prometheus-k8s-rules",
    "namespace": "vm-system"
  },
  "namespace": "vm-system",
  "name": "prometheus-k8s-rules",
  "reconcileID": "001fd51b-c200-4f85-80de-3fbc9c33f59a",
  "error": "VMRule.operator.victoriametrics.com \"prometheus-k8s-rules\" not found",
  "stacktrace": "sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem\n\tsigs.k8s.io/controller-runtime@v0.12.3/pkg/internal/controller/controller.go:273\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2\n\tsigs.k8s.io/controller-runtime@v0.12.3/pkg/internal/controller/controller.go:234"
}
```
And I don't understand, why does operator try to find deleted resource, which was deleted by operator

Chart version: 0.12.1
appVersion: 1.81.1
k8s Version: 1.19 and 1.24

**Comment by f41gh7** [comment]
Seems to be a race condition. Operator subscribes for items, `VMRules` in this case. Than it receives notification that something changed at `VMRule` and tries to load this object from kubernetes api server. But `VMRule`'s deleted by kubernetes at this moment.

 I think, this error should be suppressed.

**Comment by f41gh7** [comment]
Must be fixed at v0.28.5 version. Feel free to re-open if it not help.
